inputremapper/gui/user_interface.py

```diff
@@ -464,13 +465,18 @@ def check_macro_syntax(self):
             if not is_this_a_macro(output):
                 continue
 
-            error = parse(output, active_preset, return_errors=True)
-            if error is None:
+            error_sting = None
+            try:
+                parse(output, active_preset)
+            except MacroParsingError as error:
+                error_sting = f"{error.__class__.__name__}: {str(error)}"
+
+            if error_sting is None:
```

Comment:
I moved the exception catching out of the parse() function. And added a custom `MacroParsingError`. Now the creation of a `MacroHandler` will fail if the macro is not valid. 

Any `MappingHandler` that raises an `inputremapper.exceptions.Error` during initialization will not get added to the injection. 

TLDR: Invalid macros will now be ignored in the Injection process